### PR TITLE
Avro1.9.2 upgrade

### PR DESCRIPTION
**Pull request by junchuanwang**
Removed avro adaptor 1.4/1.6

Upgraded avro to 1.9.2 and fixed tests and SchemaTranslator logic

As a by-product, Jackson versions also upgraded to 2.10.2

### CODE REVIEW

File: data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java [hunk 1]
```diff
@@ -30,16 +30,22 @@
...
+  {
+    return jsonFromGenericRecord(record, true);
+  }
+
+  public static String jsonFromGenericRecord(GenericRecord record, boolean pretty) throws IOException
   {
     GenericDatumWriter<GenericRecord> writer = new GenericDatumWriter<>();
     ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    Encoder jsonEncoder = AvroCompatibilityHelper.newJsonEncoder(record.getSchema(), outputStream, true);
+    Encoder jsonEncoder = AvroCompatibilityHelper.newJsonEncoder(record.getSchema(), outputStream,pretty);
```
Comment (radai-rosenblatt) [inline data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java:41]: the json format has changed between avro 1.4 and 1.6. there's a variant of this method that allows you to specify the version of avro (determines the json format) - i'd set it explicitly

File: data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java [hunk 2]
```diff
@@ -2374,8 +2377,16 @@ private void testToAvroSchemaInternal(String schemaText,
             new DataToAvroSchemaTranslationOptions()
           );
           assertEquals(avroSchema2Json, avroSchema2JsonCompact);
-          Schema avroSchema2 = Schema.parse(avroSchema2Json);
-          assertEquals(avroSchema2, avroSchema);
+          Schema avroSchema2 = AvroCompatibilityHelper.parse(avroSchema2Json);
+
+
+          // TODO: remove me
```
Comment (radai-rosenblatt) [inline data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java:2383]: ?

**Review by radai-rosenblatt — APPROVED** [review]
a few nits

File: data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java (hunk 2 above)
Comment (junchuanwang) [reply data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java]: 😅

Comment (junchuanwang) [reply data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java]: thanks 😆

File: data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java (hunk 1 above)
Comment (junchuanwang) [reply data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java]: added the util method which takes AvroVersion

File: data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java [hunk 3]
```diff
@@ -31,15 +32,40 @@
...
+  }
+
+  public static String jsonFromGenericRecord(GenericRecord record, boolean pretty) throws IOException
+  {
     ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    Encoder jsonEncoder = AvroCompatibilityHelper.newJsonEncoder(record.getSchema(), outputStream, true);
+    return jsonFromGenericRecord(record,
+        outputStream,
+        AvroCompatibilityHelper.newJsonEncoder(record.getSchema(), outputStream, pretty));
+
```
Comment (nickibi) [inline data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java:44]: nit: remove extra line

File: data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java [hunk 4]
```diff
@@ -31,15 +32,40 @@
...
+  }
+
+  public static String jsonFromGenericRecord(GenericRecord record, boolean pretty, AvroVersion version) throws IOException
+  {
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    return jsonFromGenericRecord(record,
+        outputStream,
+        AvroCompatibilityHelper.newJsonEncoder(record.getSchema(), outputStream, pretty,version));
```
Comment (nickibi) [inline data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java:52]: nit: there should be a space in between "pretty, version".

**Review by nickibi — APPROVED** [review]
minor nits, otherwise LGTM

File: data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java (hunk 3 above)
Comment (junchuanwang) [reply data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java:44]: resolved

File: data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java (hunk 4 above)
Comment (junchuanwang) [reply data-avro/src/main/java/com/linkedin/data/avro/util/AvroUtil.java]: resolved

**Pull request merged**